test/videojs.ads.test.js

```diff
@@ -1,1025 +1,745 @@
-    playCount++;
+  this.player.paused = function() {
+    return false;
   };
-  player.trigger('adsready');
-  player.trigger('play');
-  player.trigger('adtimeout');
 
-  equal(1, playCount, 'play is called once');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have triggered');
-  equal(contentPlaybackReason, 'adtimeout', 'The reason for content-playback should have been adtimeout');
-});
-
-test('removes the poster attribute so it does not flash between videos', function() {
-  video.poster = 'http://www.videojs.com/img/poster.jpg';
-  ok(video.poster, 'the poster is present initially');
-
-  player.trigger('adsready');
-  player.trigger('play');
-  player.ads.startLinearAdMode();
-
-  equal(video.poster, '', 'poster is removed');
-  ok(!contentPlaybackFired, 'A content-playback event should not have triggered');
-});
-
-test('restores the poster attribute after ads have ended', function() {
-  video.poster = 'http://www.videojs.com/img/poster.jpg';
-  player.trigger('adsready');
-  player.trigger('play');
-  player.ads.startLinearAdMode();
-  player.ads.endLinearAdMode();
-
-  ok(video.poster, 'the poster is restored');
-  player.trigger('playing');
-  equal(contentPlaybackFired, 1, 'A content-playback event should have triggered');
-  equal(contentPlaybackReason, 'playing', 'The reason for content-playback should have been playing');
-});
-
-test('changing the src triggers contentupdate', function() {
-
-  // track contentupdate events
-  var contentupdates = 0;
-  player.on('contentupdate', function(){
-    contentupdates++;
-  });
+  this.player.trigger('adsready');
+  this.player.trigger('play');
+  this.clock.tick(1);
+  this.player.trigger('play');
+  this.clock.tick(1);
+  assert.strictEqual(spy.callCount, 2, 'play attempts are paused');
+});
+
+QUnit.test('stops canceling play events when an ad is playing', function(assert) {
+  assert.expect(3);
+  this.player.trigger('play');
+  this.player.trigger('adsready');
+  assert.ok(this.clock.timers.hasOwnProperty(String(this.player.ads.cancelPlayTimeout)), 'a cancel-play is scheduled');
+  this.player.trigger('adstart');
+  assert.strictEqual(this.player.ads.state, 'ad-playback', 'ads are playing');
+  assert.notOk(this.clock.timers.hasOwnProperty(String(this.player.ads.cancelPlayTimeout)), 'the cancel-play timeout is canceled');
+});
+
+QUnit.test('adstart is fired before a preroll', function(assert) {
+  assert.expect(1);
+  var spy = sinon.spy();
+  this.player.on('adstart', spy);
+  this.player.trigger('adsready');
+  this.player.trigger('play');
+  this.player.ads.startLinearAdMode();
+  assert.strictEqual(spy.callCount, 1, 'a preroll triggers adstart');
+});
+
+QUnit.test('player has the .vjs-has-started class once a preroll begins', function(assert) {
+  assert.expect(1);
+  this.player.trigger('adsready');
+
+  // This test fails unless the tech is directly triggered here. Why? Bug in
+  // video.js 5.0?
+  this.player.trigger('play');
```

Comment:
As the comment says, this test fails because this `trigger('play')` call doesn't seem to do what is expected. `this.player.tech_.trigger('play')` does work, however.